File: General/2460-apply-operations-to-an-array/main.cpp
#include <vector>
using namespace std;

class Solution {
   public:
    vector<int> applyOperations(vector<int>& nums) {
        vector<int> res(nums.size(), 0);
        for (int i = 0; i < nums.size() - 1; ++i) {
            if (nums[i] == nums[i + 1]) {
                nums[i] = 2 * nums[i];
                nums[i + 1] = 0;
            }
        }

        int j = 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i]) {
                res[j] = nums[i];
                j++;
            }
        }

        return res;
    }
};

File: General/2460-apply-operations-to-an-array/main.cs
using System;

public class Solution {
    public int[] ApplyOperations(int[] nums) {
        int[] res = new int[nums.Length];
        for (int i = 0; i < nums.Length - 1; ++i) {
            if (nums[i] == nums[i + 1]) {
                nums[i] = 2 * nums[i];
                nums[i + 1] = 0;
            }
        }

        int j = 0;
        for (int i = 0; i < nums.Length; ++i) {
            if (nums[i] != 0) {
                res[j] = nums[i];
                ++j;
            }
        }

        return res;
    }
}
